runestone/reveal/reveal.py

```diff
@@ -48,22 +48,35 @@ def visit_reveal_node(self, node):
     else:
         node.reveal_options['modaltitle'] = ''
 
-    res = TEMPLATE_START % node.reveal_options
+    if node.reveal_options['instructoronly'] and node.reveal_options['is_dynamic']:
+        res = DYNAMIC_PREFIX
+    else:
+        res = ""
+
+    res += TEMPLATE_START % node.reveal_options
     self.body.append(res)
 
 def depart_reveal_node(self,node):
 #Set options and format templates accordingly
     res = TEMPLATE_END % node.reveal_options
+    if node.reveal_options['instructoronly'] and node.reveal_options['is_dynamic']:
```

Comment:
This is repeated twice -- you might want to assign it to a variables.

Reply:
Repeated but in two different functions.  It seems more clear about what it is we are doing.